### PR TITLE
expose ability to specify prefix for signature nodes

### PR DESCRIPTION
**Pull request by wt0317**
needed to upgrade xml-crypto

see https://github.com/yaronn/xml-crypto#signedxml > computeSignature/options/prefix

### CODE REVIEW

File: lib/saml11.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ exports.create = function(options, callback) {
                   algorithms.digest[options.digestAlgorithm]);
 
   sig.signingKey = options.key;
-  
+
   sig.keyInfoProvider = {
     getKeyInfo: function () {
       return "<X509Data><X509Certificate>" + cert + "</X509Certificate></X509Data>";
```
Comment (siacomuzzi) [inline lib/saml11.js]: why namespace was not included here?

File: lib/saml20.js [hunk 2]
```diff
@@ -123,20 +124,26 @@ exports.create = function(options, callback) {
...
-  
+
   if( options.authnContextClassRef ) {
     var authnCtxClassRef = doc.getElementsByTagName('saml:AuthnContextClassRef')[0];
     authnCtxClassRef.textContent = options.authnContextClassRef;
   }
 
   var token = utils.removeWhitespace(doc.toString());
-  sig.computeSignature(token, options.xpathToNodeBeforeSignature || "//*[local-name(.)='Issuer']");
+  if (!options.location) {
```
Comment (siacomuzzi) [inline lib/saml20.js]: if `options.location` was specified without `reference` field, we should add `"//*[local-name(.)='Issuer']"` as default

File: lib/saml11.js (hunk 1 above)
Comment (wt0317) [inline lib/saml11.js]: missed it. good catch

File: lib/saml11.js [hunk 3]
```diff
@@ -169,11 +170,17 @@ function addSubjectConfirmation(options, doc, randomBytes, callback) {
 
 function sign(options, sig, doc, callback) {
   var token = utils.removeWhitespace(doc.toString());
-  sig.computeSignature(token, options.xpathToNodeBeforeSignature);
+  if (options.xpathToNodeBeforeSignature) {
+    options.location = {
```
Comment (siacomuzzi) [inline lib/saml11.js]: `option.location` is not very descriptive in the context of a saml option. Maybe we should put all fields (prefix, attrs and location) inside a unique option (like `options.signatureSettings = {}`)

File: lib/saml11.js [hunk 4]
```diff
@@ -42,10 +42,11 @@ exports.create = function(options, callback) {
                   algorithms.digest[options.digestAlgorithm]);
 
   sig.signingKey = options.key;
-  
+
   sig.keyInfoProvider = {
     getKeyInfo: function () {
-      return "<X509Data><X509Certificate>" + cert + "</X509Certificate></X509Data>";
+      var prefix = options.prefix ? options.prefix + ':' : '';
+      return "<" + prefix + "X509Data><" + prefix + "X509Certificate>" + cert + "</" + prefix + "X509Certificate></" + prefix + "X509Data>";
```
Comment (siacomuzzi) [inline lib/saml11.js]: please use `util.format` (it's more clear)

https://millermedeiros.github.io/mdoc/examples/node_api/doc/util.html#util.format

**Comment by mcastany** [comment]
We added the option `signatureNamespacePrefix` to set the signature prefix

**Pull request closed without merging**